lib/views.js

```diff
@@ -266,6 +266,63 @@ exports.places_by_type_parent_id_name = {
   }
 };
 
+exports.contacts_by_depth = {
+  map: function(doc) {
+    if (['person', 'clinic', 'health_center', 'district_hospital'].indexOf(doc.type) !== -1) {
+      var parent = doc;
+      var depth = 0;
+      while (parent) {
+        if (parent._id) {
+          emit([parent._id]);
+          emit([parent._id, depth]);
+        }
+        depth++;
+        parent = parent.parent;
+      }
+    }
+  }
+};
+
+// WARNING: If updating this function also update the getReplicationKey function in api/handlers/changes.js
+exports.docs_by_replication_key = {
+  map: function(doc) {
+    if (doc._id === 'resources' ||
+        doc._id === 'appcache' ||
+        doc.type === 'form' ||
+        doc.type === 'translations') {
+      return emit('_all');
```

Comment:
would it make sense to add `_design/medic-client` as an id check here and get rid of the hard-coding in API?